### ISSUE TITLE
[Indexer] MemberRecord::isIdentifier failes on strange use cases

### ISSUE THREAD

**Issue by camilledejoye** [phpactor/phpactor#1317]
Hi,

I just got this error:
![image](https://user-images.githubusercontent.com/11501572/131351860-4c68bae6-53f2-4b80-860d-aa23c287a609.png)

I'll grant that it's not common but yet possible:
```php
$json = json_encode(['some#hash' => 'value']);
$object = json_decode($json);
echo $object->{'some#hash'}; // Print: value
```

It was used in a test file of one of our dependencies, I simply ignored any `/vendor/**/tests/**/*` file from the index.

Although it's an extremely unusual use case, simply using `#` as a delimiter might be a bit to simplistic ?  
Low priority anyway :)

**Comment by dantleech** [comment]
yeah, should probably just skip that

**Comment by dantleech** [comment]
Fixed in #1509
